tests/phpunit/integration/Modules/Analytics_4/SettingsTest.php

```diff
@@ -90,28 +95,109 @@ function( $caps, $cap ) {
 		}
 	}
 
-	public function test_validate_google_tag_id() {
+	public function test_valid_google_tag_id() {
 		$testcase = $this->get_testcase();
 		$settings = $this->get_setting_with_owned_keys();
 		$settings->register();
 
 		$options_key = $testcase->get_option_name();
+		delete_option( $options_key );
 
+		foreach ( self::GOOGLE_TAG_IDS as $tag_id ) {
+			$options                = $settings->get();
+			$options['googleTagID'] = $tag_id;
+			$settings->set( $options );
+			$options = get_option( $options_key );
+			$testcase->assertEquals( $tag_id, $options['googleTagID'] );
+		}
+	}
+
+	public function test_invalid_google_tag_id() {
+		$testcase = $this->get_testcase();
+		$settings = $this->get_setting_with_owned_keys();
+		$settings->register();
+
+		$options_key = $testcase->get_option_name();
 		delete_option( $options_key );
 
 		$options                = $settings->get();
 		$options['googleTagID'] = 'XXX';
 		$settings->set( $options );
 		$options = get_option( $options_key );
 		$testcase->assertEquals( '', $options['googleTagID'] );
+	}
 
-		foreach ( GOOGLE_TAG_IDS as $tag_id ) {
-			$options                = $settings->get();
-			$options['googleTagID'] = $tag_id;
-			$settings->set( $options );
-			$options = get_option( $options_key );
-			$testcase->assertEquals( $tag_id, $options['googleTagID'] );
-		}
+	public function data_invalid_account_ids() {
+		return array(
+			'with invalid string' => array( 'xxxx' ),
+			'with invalid number' => array( 0, -1212 ),
+		);
+	}
+
+	/**
+	 * @dataProvider data_invalid_account_ids
+	 */
+	public function test_invalid_tag_account_id( $invalid_id ) {
```

Comment:
Thanks, @sashadoes. Code looks good to me, but I think we can combine all `test_invalid_*` and `test_valid_*` methods into one method with the right `dataProvider`. Could you please update it?